src/git-prune-branches

```diff
@@ -22,8 +22,23 @@ function should_delete_branch {
 }
 
 
+function set_previous_branch {
+  # previous branch was deleted
+  if [ "$prune_branches_initial_previous_branch_deleted" = true ]; then
+    set_branch_as_previous_branch "$MAIN_BRANCH_NAME"
+  # current branch was deleted
+  elif [ "$prune_branches_initial_branch_deleted" = true ]; then
+    set_branch_as_previous_branch "$INITIAL_PREVIOUS_BRANCH_NAME"
+  # some other branch was deleted
+  else
+    set_branch_as_previous_branch "$INITIAL_PREVIOUS_BRANCH_NAME"
+  fi
+}
+
+
 function steps {
-  local initial_branch_deleted=false
+  prune_branches_initial_branch_deleted=false
+  prune_branches_initial_previous_branch_deleted=false
```

Comment:
It may be used in another git town command but shouldn't be used in any helper. Also what about the idea of instead having the global PREVIOUS_BRANCH_NAME simply be modified anytime we need to set the pervious branch to something other than its initial value.